llm/llm.go

```diff
@@ -48,28 +39,56 @@ func Prompt(ctx context.Context, config Config, systemPrompt string, promptParts
 		content = append(content, llms.TextParts(llms.ChatMessageTypeHuman, p))
 	}
 
-	resp, err := model.GenerateContent(ctx, content, llms.WithTemperature(0))
-	if err != nil {
-		return "", nil, fmt.Errorf("failed to generate response: %w", err)
-	}
-
-	if len(resp.Choices) == 0 {
-		return "", nil, errors.New("no response from LLM")
-	}
-
-	content = append(content, llms.TextParts(llms.ChatMessageTypeAI, resp.Choices[0].Content))
-	return resp.Choices[0].Content, content, nil
+	return PromptWithHistory(ctx, config, content, "")
 }
 
-func PromptWithHistory(ctx context.Context, config Config, history []llms.MessageContent, prompt string) (string, []llms.MessageContent, error) {
+func PromptWithHistory(ctx dutyctx.Context, config Config, history []llms.MessageContent, prompt string) (string, []llms.MessageContent, error) {
 	model, err := getLLMModel(ctx, config)
 	if err != nil {
 		return "", nil, err
 	}
 
-	content := append(history, llms.TextParts(llms.ChatMessageTypeHuman, prompt))
+	messages := append(history, llms.TextParts(llms.ChatMessageTypeHuman, prompt))
 
-	resp, err := model.GenerateContent(ctx, content, llms.WithTemperature(0))
+	options := []llms.CallOption{llms.WithTemperature(0)}
+
+	// Add backend-specific options for tool choice
+	switch config.Backend {
+	case api.LLMBackendOpenAI:
+		// Do nothing
+		// NOTE: we use `response_format` instead of function calling & that's configured during model creation not when prompting.
+		// OpenAI does support function calling, but I don't think we can force the model to use that tool
+		// like we can with Anthropic.
+
+	case api.LLMBackendGemini:
+		// Do nothing
```

Comment:
According to https://pkg.go.dev/google.golang.org/genai#GenerateContentConfig there is 
```
// Schema that the generated candidate text must adhere to.
	ResponseSchema *[Schema](https://pkg.go.dev/google.golang.org/genai#Schema) `json:"responseSchema,omitempty"`
```

Reply:
hmm, we are using https://github.com/google/generative-ai-go/genai
which doesn't seem to have that config. We do it via tools.

But googleapis/go-genai seems to be the new repo for it (based on the commits)
Will migrate to that.

Reply:
https://github.com/flanksource/mission-control/pull/1938/files#diff-cfdf50fee4d1c92bc2e7bfb32ce2c3fa7b2f53f4a8d0c95b9d6883e4b8c8c535R140-R148